### PR TITLE
Micrometer POC

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: custom/src/main/java/com/splunk/opentelemetry/metrics/MicrometerInstaller.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        new SignalFxMeterRegistry(new SplunkMetricsConfig(), Clock.SYSTEM);
+    NamingConvention signalFxNamingConvention = signalFxRegistry.config().namingConvention();
+    signalFxRegistry.config().namingConvention(new OtelNamingConvention(signalFxNamingConvention));
+
+    // TODO: agent does not respond?
+    try {
+      Logger errorLogger = LoggerFactory.getLogger(MicrometerInstaller.class);
+
+      Field exceptionListeners =
+          SignalFxMeterRegistry.class.getDeclaredField("onSendErrorHandlerCollection");
```
Comment (jkwatson) [inline custom/src/main/java/com/splunk/opentelemetry/metrics/MicrometerInstaller.java:35]: there's no better way to do this? What does the built-in listener do right now?

File: instrumentation/servlet/src/main/java/com/splunk/opentelemetry/servlet/ServletInstrumentationModule.java [hunk 2]
```diff
@@ -110,4 +122,14 @@ public void set(HttpServletResponse carrier, String key, String value) {
       carrier.addHeader(key, value);
     }
   }
+
+  public static final class RequestCounter {
+    private static final Counter REQUEST_COUNT = Metrics.counter("http.request.count");
```
Comment (jkwatson) [inline instrumentation/servlet/src/main/java/com/splunk/opentelemetry/servlet/ServletInstrumentationModule.java:127]: Can we add an example here of adding a tag to the counter? I don't remember how the micrometer APIs work here, but it'll be an important thing to be able to do. maybe just the HTTP verb for starters?

File: custom/src/main/java/com/splunk/opentelemetry/metrics/MicrometerInstaller.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply custom/src/main/java/com/splunk/opentelemetry/metrics/MicrometerInstaller.java:35]: The built-in listener doesn't log the stacktrace. There's no setter or anything that allows providing your own listener, so I  had to force my way 🤷‍♂️

File: instrumentation/servlet/src/main/java/com/splunk/opentelemetry/servlet/ServletInstrumentationModule.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply instrumentation/servlet/src/main/java/com/splunk/opentelemetry/servlet/ServletInstrumentationModule.java]: Added 👍

File: testing/agent-metrics/src/main/java/com/splunk/opentelemetry/testing/TestMetricsInstaller.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+package com.splunk.opentelemetry.testing;
+
+import com.google.auto.service.AutoService;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import io.opentelemetry.javaagent.spi.ComponentInstaller;
+
+@AutoService(ComponentInstaller.class)
+public class TestMetricsInstaller implements ComponentInstaller {
```
Comment (breedx-splk) [inline testing/agent-metrics/src/main/java/com/splunk/opentelemetry/testing/TestMetricsInstaller.java:25]: Rename to `TestMicrometerInstaller` to match the true impl name?

File: custom/src/main/java/com/splunk/opentelemetry/metrics/SplunkMetricsConfig.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package com.splunk.opentelemetry.metrics;
```
Comment (breedx-splk) [inline custom/src/main/java/com/splunk/opentelemetry/metrics/SplunkMetricsConfig.java:1]: new file missing copyright.

File: custom/src/main/java/com/splunk/opentelemetry/metrics/OtelNamingConvention.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package com.splunk.opentelemetry.metrics;
```
Comment (breedx-splk) [inline custom/src/main/java/com/splunk/opentelemetry/metrics/OtelNamingConvention.java:1]: copyright notice

File: custom/src/main/java/com/splunk/opentelemetry/metrics/MicrometerInstaller.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
+package com.splunk.opentelemetry.metrics;
```
Comment (breedx-splk) [inline custom/src/main/java/com/splunk/opentelemetry/metrics/MicrometerInstaller.java:1]: copyright notice

File: bootstrap/src/main/java/com/splunk/opentelemetry/javaagent/bootstrap/MetricsBridge.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package com.splunk.opentelemetry.javaagent.bootstrap;
```
Comment (breedx-splk) [inline bootstrap/src/main/java/com/splunk/opentelemetry/javaagent/bootstrap/MetricsBridge.java:1]: copyright notice

File: bootstrap/src/main/java/com/splunk/opentelemetry/javaagent/bootstrap/GlobalMetricsTags.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
+package com.splunk.opentelemetry.javaagent.bootstrap;
```
Comment (breedx-splk) [inline bootstrap/src/main/java/com/splunk/opentelemetry/javaagent/bootstrap/GlobalMetricsTags.java:1]: copyright notice.

File: custom/src/main/java/com/splunk/opentelemetry/metrics/SplunkMetricsConfig.java (hunk 4 above)
Comment (breedx-splk) [reply custom/src/main/java/com/splunk/opentelemetry/metrics/SplunkMetricsConfig.java:1]: ok cool.  was just setting the stage, but I get it.

**Pull request closed without merging**